collector/src/bin/rustc-perf-collector/execute.rs

```diff
@@ -348,346 +605,94 @@ impl Benchmark {
         }
     }
 
-    /// Run a specific benchmark under perf-stat.
+    /// Run a specific benchmark under a processor + profiler combination.
     pub fn measure(
         &self,
+        processor: &mut dyn Processor,
```

Comment:
Let's refactor this to take `Processor` generically, that is this would be `&mut T where T: Processor`. (If I understand the callsites correctly this should be a trivial change, if it isn't then we don't need to do this).

Reply:
The change works, but since I store a `dyn Processor` reference elsewhere, I used `dyn Processor` here for consistency. Seem reasonable?

Reply:
Makes sense, I had missed that before.